### PR TITLE
Fix last remnants of global url and ua usage

### PR DESCRIPTION
**Pull request by aaannz**

### CODE REVIEW

**Comment by aaannz** [comment]
WIP are tests for asset upload

File: lib/OpenQA/Worker/Jobs.pm [hunk 1]
```diff
@@ -252,10 +252,10 @@ sub upload {
         print $log "Checksum $csum1:$csum2 Sizes:$size1:$size2\n";
         close $log;
         if ($csum1 eq $csum2 && $size1 eq $size2) {
-            my $ua_url = $OpenQA::Worker::Common::url->clone;
+            my $ua_url = $hosts->{$current_host}{url}->clone;
             $ua_url->path("jobs/$job_id/ack_temporary");
 
-            $OpenQA::Worker::Common::ua->post($ua_url => form => {temporary => $res->res->json->{temporary}});
+            $hosts->{$current_host}{ua}->post($ua_url => form => {temporary => $res->res->json->{temporary}});
```
Comment (foursixnine) [inline lib/OpenQA/Worker/Jobs.pm]: I was just doing a similar approach using api_call

**Comment by coolo** [comment]
0.00% of diff hit

**Comment by coolo** [comment]
10:20:31.8605 26119 ERROR: Machine not shut down when uploading disks!

we need to extend the tests for that to work. Too bad the tests are in os-autoinst :(

**Comment by aaannz** [comment]
This requires https://github.com/os-autoinst/os-autoinst/pull/709 and https://github.com/os-autoinst/openQA/pull/1203 to pass upload test.

**Comment by aaannz** [comment]
rebased

**Pull request merged**